plottr/node/dim_reducer.py

```diff
@@ -243,38 +248,38 @@ def getRoles(self) -> Dict[str, Dict[str, Any]]:
         return ret
 
     @Slot(str, str)
-    def setDimInfo(self, dim: str, info: str = ''):
+    def setDimInfo(self, dim: str, info: str = '') -> None:
         try:
             item = self.findItems(dim, QtCore.Qt.MatchExactly, 0)[0]
             item.setText(3, info)
         except IndexError:
             pass
 
     @Slot(dict)
-    def setDimInfos(self, infos: Dict[str, str]):
+    def setDimInfos(self, infos: Dict[str, str]) -> None:
         for ax, info in infos.items():
             self.setInfo(ax, info)
 
 
 class DimensionReductionAssignmentWidget(DimensionAssignmentWidget):
 
-    def __init__(self, parent=None):
+    def __init__(self, parent: Optional[QtWidgets.QWidget] = None):
         super().__init__(parent)
 
         self.availableChoices[MeshgridDataDict] += [
             ReductionMethod.average.value,
             ReductionMethod.elementSelection.value,
         ]
 
-    def getRole(self, name: str):
+    def getRole(self, name: str) -> Tuple[str, Dict]:
         role, opts = super().getRole(name)
 
         if role == ReductionMethod.elementSelection.value:
             opts['index'] = self.choices[name]['optionsWidget'].value()
 
         return role, opts
 
-    def setRole(self, dim: str, role: str = None, **kw):
+    def setRole(self, dim: str, role: str = None, **kw: Any) -> None:
```

Comment:
why does mypy not flag that an "str" argument was assigned "NoneType" value?
```suggestion
    def setRole(self, dim: str, role: Optional[str] = None, **kw: Any) -> None:
```

Reply:
Because any argument that has a not optional type but an optional default value is implicitly optional unless you tell mypy otherwise https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-optional